gymnasium/dev_wrappers/utils/reshape_space.py

```diff
@@ -0,0 +1,43 @@
+"""A set of utility functions for lambda wrappers."""
+import typing
+from functools import singledispatch
+from typing import Any, Callable
+
+import jumpy as jp
+
+from gymnasium.dev_wrappers import FuncArgType
+from gymnasium.error import InvalidSpaceOperation
+from gymnasium.spaces import Box, Discrete, MultiBinary, MultiDiscrete, Space
+
+
+@singledispatch
+def reshape_space(space: Space, args: FuncArgType, fn: Callable) -> Any:
+    """Reshape space with the provided args."""
+    raise NotImplementedError
+
+
+@reshape_space.register(Discrete)
+@reshape_space.register(MultiBinary)
+@reshape_space.register(MultiDiscrete)
+def _reshape_space_not_reshapable(space, args: FuncArgType, fn: Callable):
+    """Return original space shape for not reshable space.
+
+    Trying to reshape `Discrete`, `Multibinary` and `MultiDiscrete`
+    spaces has no effect.
+    """
+    if args:
+        raise InvalidSpaceOperation(f"Cannot reshape a space of type {type(space)}.")
+    return space
+
+
+@reshape_space.register(Box)
+def _reshape_space_box(space, args: FuncArgType[typing.Tuple[int, int]], fn: Callable):
```

Comment:
`typing.Tuple[int,...]` for x-D Box shapes

Reply:
https://github.com/Farama-Foundation/Gymnasium/pull/22/commits/de9e43acdbe40a85cfdf6f0a39dce7a1326b1a03